client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,45 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy {@link RetentionPolicy} is set to 'CONSUMPTION' based,
+     * the Reader Group needs to register with the Stream as a Subscriber
```

Comment:
`the Reader Group needs to register with the Stream as a Subscriber`
This is simply not true.

Reply:
It is true, but an implementation detail. Changed now.

Reply:
It is perfectly valid to set `no_impact` and not be a subscriber. So it's not needed.